ingenialink/dictionary.py

```diff
@@ -51,7 +178,148 @@ def __init__(self, dictionary_path):
         """Categories: Instance of all the categories in the dictionary."""
         self.errors = None
         """Errors: Instance of all the errors in the dictionary."""
+        self._registers = []
+        """Registers: Instance of all the registers in the dictionary"""
 
-    @abstractmethod
     def registers(self, subnode):
-        raise NotImplementedError
+        """Gets the register dictionary to the targeted subnode.
+
+        Args:
+            subnode (int): Identifier for the subnode.
+
+        Returns:
+            dict: Dictionary of all the registers for a subnode.
+
+        """
+        return self._registers[subnode]
+
+    def read_dictionary(self):
+        """Reads the dictionary file and initializes all its components."""
+        try:
+            with open(self.path, 'r', encoding='utf-8') as xdf_file:
+                tree = ET.parse(xdf_file)
+        except FileNotFoundError:
+            raise FileNotFoundError(f"There is not any xdf file in the path: {self.path}")
+        root = tree.getroot()
+
+        device = root.find(DICT_ROOT_DEVICE)
+
+        # Subnodes
+        if root.findall(DICT_ROOT_AXES):
+            self.subnodes = len(root.findall(DICT_ROOT_AXIS))
+
+        for _ in range(self.subnodes):
+            self._registers.append({})
+
+        # Categories
+
+        self.categories = DictionaryCategories(root)
+
+        # Errors
+        self.errors = DictionaryErrors(root)
+
+        # Version
+        version_node = root.find(DICT_ROOT_VERSION)
+        if version_node is not None:
+            self.version = version_node.text
+
+        self.firmware_version = device.attrib.get('firmwareVersion')
+        product_code = device.attrib.get('ProductCode')
+        if product_code is not None and product_code.isdecimal():
+            self.product_code = int(product_code)
+        self.part_number = device.attrib.get('PartNumber')
+        revision_number = device.attrib.get('RevisionNumber')
+        if revision_number is not None and revision_number.isdecimal():
+            self.revision_number = int(revision_number)
+        self.interface = device.attrib.get('Interface')
+
+        if root.findall(DICT_ROOT_AXES):
+            # For each axis
+            for axis in root.findall(DICT_ROOT_AXIS):
+                for register in axis.findall(DICT_REGISTERS_REGISTER):
+                    current_read_register = self._read_register(register)
+                    if current_read_register:
+                        self._add_register_list(current_read_register)
+        else:
+            for register in root.findall(DICT_ROOT_REGISTER):
+                current_read_register = self._read_register(register)
+                if current_read_register:
```

Comment:
I check it because if there is a lack of defined register's attribute inside a xdf file, this register won't be added into the list. I'm going to add a "return None" for these cases.

Reply:
Ok, add in the docstring when it returns None